index.js

```diff
@@ -1,15 +1,25 @@
 'use strict';
+var errorEx = require('error-ex');
 var fallback = require('./vendor/parse');
 
+var JSONError = errorEx('JSONError', {
+	fileName: errorEx.append('in %s')
+});
+
 module.exports = function (x, reviver) {
 	try {
-		return JSON.parse(x, reviver);
-	} catch (err) {
-		fallback.parse(x, {
-			mode: 'json',
-			reviver: reviver
-		});
+		try {
+			return JSON.parse(x, reviver);
+		} catch (err) {
+			fallback.parse(x, {
+				mode: 'json',
+				reviver: reviver
+			});
 
+			throw err;
+		}
+	} catch (err) {
+		JSONError.call(err);
 		throw err;
```

Comment:
I guess I'm used to doing it when creating gulp plugins: https://github.com/sindresorhus/gulp-esformatter/blob/894ead7dd5b796e26e8d3c74cdf0e4bcc55db76a/index.js#L22

https://github.com/gulpjs/gulp-util/blob/master/lib/PluginError.js

It feels more natural just passing the existing error into a function and getting a new one out, instead of mutating the error in-place.